### ISSUE TITLE
Thread team strategies

### ISSUE THREAD

**Issue by MathGaron** [RoboCupULaval/PythonFramework#7]
The teams play on the same thread and affect the other teams depending on their timing etc. We should run each team on a thread

**Comment by felixpelletier23** [comment]
Would a Python thread work in this case?

**Comment by MathGaron** [comment]
yes, the system will have to be architectured in a different way.. when we think of it, normally the engine should execute only one strategie at the time.. but we dont for the competitions..

**Comment by MathGaron** [comment]
A first implementation has been push on a feature branch, we need to test before going on
